src/mmw/mmw/middleware.py

```diff
@@ -0,0 +1,40 @@
+from django.conf import settings
+from django.core.handlers.base import BaseHandler
+from django.core import urlresolvers
+
+
+def bypass_middleware(view):
+    view.bypass_middleware = True
+
+    return view
+
+
+class BypassMiddleware(object):
+    """
+    Customized version of a gist detailing this technique by @bryanhelmig
+
+    See also: https://gist.github.com/bryanhelmig/9d09d1bd9a63504371d2
+    """
+    def process_request(self, request):
+        """Replicates a lot of code from BaseHandler#get_response."""
+        # Setup URL resolver
+        urlconf = settings.ROOT_URLCONF
+        urlresolvers.set_urlconf(urlconf)
+        resolver = urlresolvers.RegexURLResolver(r'^/', urlconf)
+        callback, callback_args, \
+            callback_kwargs = resolver.resolve(request.path_info)
+
+        if getattr(callback, 'bypass_middleware', False):
+            # bypass_middleware decorator was used; zero out all
+            # middleware and return the response.
+            handler = BaseHandler()
+
+            handler._request_middleware = []
+            handler._view_middleware = []
+            handler._template_response_middleware = []
+            handler._response_middleware = []
+            handler._exception_middleware = []
+
+            response = handler.get_response(request)
+
+            return response
```

Comment:
I may be missing something here, but what other args on `process_request`?

If I try to intercept at `process_view` and conditionally call `view_func`, the database connectivity exception occurs during or after `view_func`:

```python
def process_view(self, request, view_func, view_args, view_kwargs):
    health_check_func = '.'.join([view_func.__module__,
                                  view_func.func_name])

    if health_check_func == 'apps.monitoring.views.health_check':
        print("joker")
        return view_func(request, *view_args, **view_kwargs)
```

In the logs, I see:

```
joker
Internal Server Error: /health-check/
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 223, in get_response
    response = middleware_method(request, response)
  File "/usr/local/lib/python2.7/dist-packages/debug_toolbar/middleware.py", line 94, in process_response
    new_response = panel.process_response(request, response)
  File "/usr/local/lib/python2.7/dist-packages/debug_toolbar/panels/request.py", line 55, in process_response
    for k in sorted(request.session.keys(), key=force_text)]
  File "/usr/local/lib/python2.7/dist-packages/django/contrib/sessions/backends/base.py", line 119, in keys
    return self._session.keys()
  File "/usr/local/lib/python2.7/dist-packages/django/contrib/sessions/backends/base.py", line 181, in _get_session
    self._session_cache = self.load()
  File "/usr/local/lib/python2.7/dist-packages/django/contrib/sessions/backends/db.py", line 21, in load
    expire_date__gt=timezone.now()
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/manager.py", line 127, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 328, in get
    num = len(clone)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 144, in __len__
    self._fetch_all()
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 965, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 238, in iterator
    results = compiler.execute_sql()
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/sql/compiler.py", line 838, in execute_sql
    cursor = self.connection.cursor()
  File "/usr/local/lib/python2.7/dist-packages/debug_toolbar/panels/sql/tracking.py", line 42, in cursor
    return state.Wrapper(connection._djdt_cursor(), connection, panel)
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/base/base.py", line 162, in cursor
    cursor = self.make_debug_cursor(self._cursor())
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/base/base.py", line 135, in _cursor
    self.ensure_connection()
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/base/base.py", line 130, in ensure_connection
    self.connect()
  File "/usr/local/lib/python2.7/dist-packages/django/db/utils.py", line 97, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/base/base.py", line 130, in ensure_connection
    self.connect()
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/base/base.py", line 119, in connect
    self.connection = self.get_new_connection(conn_params)
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/postgresql_psycopg2/base.py", line 176, in get_new_connection
    connection = Database.connect(**conn_params)
  File "/usr/local/lib/python2.7/dist-packages/psycopg2/__init__.py", line 164, in connect
    conn = _connect(dsn, connection_factory=connection_factory, async=async)
OperationalError: could not connect to server: Connection refused
	Is the server running on host "33.33.34.30" and accepting
	TCP/IP connections on port 5432?
```